### ISSUE TITLE
PR2492 causes regression to Modal

### ISSUE THREAD

**Issue by yiwen101** [MarkBind/markbind#2508]
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Tell us about your environment

MacOs

### MarkBind version

Current master branch

### Describe the bug and the steps to reproduce it

After #2492, modal cannot pop up properly.
Can test with this: 

```
More about <trigger for="modal:loremipsum">trigger</trigger>.
<modal header="**Modal header** :rocket:" id="modal:loremipsum">
  Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore
  magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
  consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</modal>
<br>
This is the same <trigger for="modal:loremipsum">trigger</trigger> as last one.

<trigger for="modal:centered">This is a trigger for a centered modal</trigger>.

<modal header="**Centered** :rocket:" id="modal:centered" center>
  Centered
</modal>

<trigger for="modal:ok-text">This is a trigger for a modal with a custom OK button</trigger>.

<modal header="OK button visible!" id="modal:ok-text" ok-text="Custom OK">
  Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore
  magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
  consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</modal>
```

**Comment by damithc** [comment]
Thanks for reporting this @yiwen101 
If confirmed, need to fix quickly, as we can't do another release until this is fixed.

**Comment by yucheng11122017** [comment]
Will revert the PR first. @LamJiuFong would be good if you have the bandwidth to figure out the issue and if its possible to fix.

**Comment by LamJiuFong** [comment]
@yucheng11122017  I am looking at it now, thanks

**Comment by LamJiuFong** [comment]
This was the exact issue:
<img width="380" alt="Screenshot 2024-04-08 at 6 50 38 PM" src="https://github.com/MarkBind/markbind/assets/122192553/f8d465f8-aa7f-48f7-a802-cd63844064c0">

I believe the issue occurred due to the dependencies between this script and the deferred scripts:
<img width="634" alt="Screenshot 2024-04-08 at 6 51 44 PM" src="https://github.com/MarkBind/markbind/assets/122192553/ffda8f3d-7ec7-4ada-a40c-91858de6975b">

<img width="441" alt="Screenshot 2024-04-08 at 6 45 39 PM" src="https://github.com/MarkBind/markbind/assets/122192553/c609a8a2-4356-4aa6-b7b2-f215f15ee084">

Once the scripts in the second image are deferred, they are executed later than the first one, hence there is a reference error.

Solution:
We have to make sure that the four scripts are being executed before the first one, in this case, we cannot use `defer` because
i. deferred scripts are loaded last (as stated above)
ii. While `defer` preserves execution order, adding `defer` to the first script component still cannot work because `defer` only works on `<script>` with a `src` attribute.

However, we can still move the scripts to the bottom (have to make sure they are above the first script).
It can still save rendering time as the files are executed after all DOM elements are loaded.

**Comment by yucheng11122017** [comment]
> This was the exact issue: <img alt="Screenshot 2024-04-08 at 6 50 38 PM" width="380" src="https://private-user-images.githubusercontent.com/122192553/320452846-f8d465f8-aa7f-48f7-a802-cd63844064c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2MzIzMTEsIm5iZiI6MTcxMjYzMjAxMSwicGF0aCI6Ii8xMjIxOTI1NTMvMzIwNDUyODQ2LWY4ZDQ2NWY4LWFhN2YtNDhmNy1hODAyLWNkNjM4NDQwNjRjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOVQwMzA2NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZWIzMTcxMDIzYjlhMzEzYmYxYjQ0YjhhNzkwNTAyMzE2ZmU2ZDdmZGNjYWE3ZDg1MDMwNmFlN2RmNTFmMmI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NKV8wEd07YEEu84DbNIj8KVPDjXKHRuiXLT6YgIkaK8">
> 
> I believe the issue occurred due to the dependencies between this script and the deferred scripts: <img alt="Screenshot 2024-04-08 at 6 51 44 PM" width="634" src="https://private-user-images.githubusercontent.com/122192553/320453148-ffda8f3d-7ec7-4ada-a40c-91858de6975b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2MzIzMTEsIm5iZiI6MTcxMjYzMjAxMSwicGF0aCI6Ii8xMjIxOTI1NTMvMzIwNDUzMTQ4LWZmZGE4ZjNkLTdlYzctNGFkYS1hNDBjLTkxODU4ZGU2OTc1Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOVQwMzA2NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02OGZiMjJmYWMyNzc0MmNlYmNhZjliZTFiZjZkNmRmYjc4ZjcxMTc3ZDAwNzIzOGNjNGU5YmFkMDk3YTQ5MmIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TYEyXUjBi0gqBFAFgOOnva4RmzSiSBVVCDCVswRKke8">
> 
> <img alt="Screenshot 2024-04-08 at 6 45 39 PM" width="441" src="https://private-user-images.githubusercontent.com/122192553/320451446-c609a8a2-4356-4aa6-b7b2-f215f15ee084.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2MzIzMTEsIm5iZiI6MTcxMjYzMjAxMSwicGF0aCI6Ii8xMjIxOTI1NTMvMzIwNDUxNDQ2LWM2MDlhOGEyLTQzNTYtNGFhNi1iN2IyLWYyMTVmMTVlZTA4NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOVQwMzA2NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMmJjNTVlNTMxYjRiYWZiZmY1YzY1ZDFiNWRiNTIyZDBiZTNhZDM3MzQxN2ExZDQ0Y2ZkNTQ4N2ZiNDI4NTIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eV4ZKo6O7zEFGh_leNyiGUFfFIyQYEM4QF3FVX3N3DE">
> Once the scripts in the second image are deferred, they are executed later than the first one, hence there is a reference error.
> 
> Solution: We have to make sure that the four scripts are being executed before the first one, in this case, we cannot use `defer` because i. deferred scripts are loaded last (as stated above) ii. While `defer` preserves execution order, adding `defer` to the first script component still cannot work because `defer` only works on `<script>` with a `src` attribute.
> 
> However, we can still move the scripts to the bottom (have to make sure they are above the first script). It can still save rendering time as the files are executed after all DOM elements are loaded.

Thanks for the quick investigation @LamJiuFong! 

Is it possible to change the first script to use a `src` attribute? 

Could you investigate on the performance difference when pushing the scripts to the bottom vs our current implementation?

**Comment by LamJiuFong** [comment]
@yucheng11122017 thanks for the suggestions, I have tried them out:

> Could you investigate on the performance difference when pushing the scripts to the bottom vs our current implementation?

On my end, pushing the scripts to the bottom takes ~17s to load the DOM content while the current implementation takes ~20s to load

> Is it possible to change the first script to use a `src` attribute?

I think this will be tricky as we have to add js files as one of the `asset` of a `PageConfig` when creating new pages in `Site/index.ts::createPage`, it would probably involve some refactoring of code

Perhaps just moving the scripts to the bottom is enough, what do you think?

**Comment by kaixin-hc** [comment]
@LamJiuFong I think it is fine to move the scripts to the bottom to fix the OG issue; then we can change this issue to be about exploring refactoring / making it possible to use defer, with more investigation.
